internal/controller/argocd.go

```diff
@@ -150,6 +150,29 @@ func (clctrl *ClusterController) InitializeArgoCD() error {
 	return nil
 }
 
+func RestartDeployment(ctx context.Context, clientset kubernetes.Interface, namespace string, deployment_name string) error {
```

Comment:
Complete and utter nitpick, but two things:

* Go recommends to use camel case (`camelCase`) variables instead of snake case (`snake_case`)
* And in Go you can "merge" two variable declarations if the type is the same

So for the first one, the `deployment_name` variable should be instead `deploymentName`.

For the second one, you can rewrite your current function into this:

```suggestion
func RestartDeployment(ctx context.Context, clientset kubernetes.Interface, namespace, deploymentName string) error {
```